airesources/Haskell/Hlt/Entity.hs

```diff
@@ -0,0 +1,113 @@
+-- | Defines Planets and Ships and various functions that operate on them.
+module Hlt.Entity where
+
+import qualified Data.Map as Map
+import Hlt.Constants
+
+-- | Grouping of a player (bot) and its corresponding Ships.
+data Player = Player { playerId :: Int, ships :: Map.Map Int Entity } deriving (Show)
+
+{-|
+  An Entity is an object on the GameMap.
+
+  All Entities have an (x, y) position and a radius.
+
+  Planets and Ships both have corresponding health and owner attributes and various extra attributes.
+-}
+data Entity = Location { x :: Float
+                       , y :: Float
+                       , radius :: Float
+                       }
+            | Planet   { entityId :: Int
+                       , x :: Float
+                       , y :: Float
+                       , health :: Int
+                       , radius :: Float
+                       , dockingSpots :: Int
+                       , production :: Int
+                       , owner :: Int
+                       , dockedShips :: [Int]
+                       }
+            | Ship     { entityId :: Int
+                       , x :: Float
+                       , y :: Float
+                       , health :: Int
+                       , radius :: Float
+                       , owner :: Int
+                       , status :: Int
+                       , planet :: Int
+                       , dockingProgress :: Int
+                       , weaponCooldown :: Int
+                       }
+            deriving (Show)
+
+-- | Entities are equal if their locations on the GameMap are equal.
+instance Eq Entity where
+   a == b = (x a == x b && y a == y b)
+
+-- | Returns the x component of the difference between two Entities.
+dX :: Entity -> Entity -> Float
+dX a b = x b - x a
+
+-- | Returns the y component of the difference between two Entities.
+dY :: Entity -> Entity -> Float
+dY a b = y b - y a
+
+-- | Returns the angle between two Entities.
+getAngle :: Entity -> Entity -> Float
```

Comment:
I suggest you make it explicit if angle is radians or degrees throughout the kit. For example this might be getAngleDeg or getAngleRad.